java/gradle/library.gradle

```diff
@@ -85,44 +31,20 @@ task javadocJar(type: Jar, dependsOn: javadoc) {
     destinationDir = new File("${libDir}")
 }
 
-artifacts {
-    archives jar, jarSources, javadocJar
-}
-
-signing {
-    required {
-        gradle.taskGraph.hasTask("signArchives") ||
-        gradle.taskGraph.hasTask("uploadArchives") ||
-        ( project.hasProperty('signing.keyId') &&
-          project.hasProperty('signing.password') &&
-          project.hasProperty('signing.secretKeyRingFile') )
-    }
-    sign configurations.archives
+// The maven plugin change in Gradle 7.3, for compatibility with older gradle versions
+// we split maven related tasks to separate files that use the corresponding builtin gradle
+// plug-ins
+project.ext.pomName = "${libDir}/${project.name}-${project.version}.pom"
+if(GradleVersion.current() >= GradleVersion.version('7.3')) {
+    apply from: "$project.ext.topSrcDir/java/gradle/maven-publish.publishing.gradle"
+} else {
+    apply from: "$project.ext.topSrcDir/java/gradle/maven.publishing.gradle"
 }
 
-ext.mavenRepo = rootProject.hasProperty('mavenRepo')?project.mavenRepo:System.getenv('MAVEN_REPO')?:''
-ext.mavenSnapshotRepo = rootProject.hasProperty('mavenSnapshotRepo')?project.mavenSnapshotRepo:System.getenv('MAVEN_SNAPSHOT_REPO')?:''
-ext.mavenUser = rootProject.hasProperty('mavenUser')?project.mavenUser:System.getenv('MAVEN_USER')?:''
-ext.mavenPassword = rootProject.hasProperty('mavenPassword')?project.mavenPassword:System.getenv('MAVEN_PASSWORD')?:''
```

Comment:
Do we not need this configuration settings at all now?

Reply:
AFAIK we don't use that for our maven upload.  We have a separate dist-util job for uploading maven artifacts.

Reply:
Ok, we used to use this, but not anymore?